inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -19,78 +18,254 @@
 class SqliteManager:
     def __init__(self, db_file=UDM_DB_FILE) -> None:
         self._db_file = db_file
+        self._conn: sqlite3.Connection = sqlite3.connect(self._db_file)
+        self._create_tables_if_not_exist()
+        self._cursor = self._conn.cursor()
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Implement context managers for database connections to ensure they are always properly closed.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        self._conn = None
        try:
            self._conn = sqlite3.connect(self._db_file)
            self._cursor = self._conn.cursor()
        except sqlite3.Error as e:
            logger.error(f"Unable to connect to database: {e}")
```